src/plugins/action-queue/cleaners.js

```diff
@@ -0,0 +1,164 @@
+// @flow
+
+//
+// Action Operations
+//
+import { type Cleaner, asArray, asCodec, asEither, asMaybe, asNumber, asObject, asOptional, asString, asValue } from 'cleaners'
+
+import { type ActionEffect, type ActionOp, type ActionProgram, type ActionProgramState } from './types'
+
+// A serializeable error object
+const asJsonError = asObject({
+  name: asString,
+  message: asString
+}).withRest
+
+const asError = asCodec(
+  (raw: any) => {
+    // Handle Error
+    if (raw instanceof Error) return raw
+
+    // Handle JsonError
+    const jsonError = asMaybe(asJsonError)(raw)
+    if (jsonError != null) {
+      const error = new Error(jsonError.message)
+      Object.defineProperty(error, 'name', {
+        value: jsonError.name
+      })
+      return error
+    }
+
+    // Invalid type
+    throw new TypeError('Expected Error')
+  },
+  // Serialize as a JsonError
+  asJsonError
+)
+
+export const asActionOp: Cleaner<ActionOp> = asEither(
```

Comment:
Not in the scope of this review, but this is currently a Flow error. Does there need to be a change to package.json to support the extension of asEither?

Reply:
Disregard, just heard you have a cleaner PR out.